File: P1A9_Empleados/C++/Archivos_Origen/testEmp.cpp
#include "../Archivos_Encabezado/empleado.h"
#include <iostream>
using namespace std;

int main(){
    empleado e1;
    empleado e2("Juan", 2022, 1000);
    cout << "------------" << endl;
    cout << e1.toString() << endl;
    cout << "------------" << endl;
    cout << e2.toString() << endl;
    cout << "------------" << endl;

    //Solicitar datos al usuario
    string nombre;
    int anioDeContratacion;
    double salario;

    cout << "Ingrese el nombre: ";
    getline(cin, nombre);

    cout << "Ingrese el anio de contratacion: ";
    cin >> anioDeContratacion;

    cout << "Ingrese el salario: ";
    cin >> salario;

    cout << "------------" << endl;
    empleado e3(nombre, anioDeContratacion, salario);
    cout << e3.toString() << endl;
    cout << "------------" << endl;

    return 0;
}

File: P1A9_Empleados/C#/Empleado/testEmp.cs
using System;

class testEmpleado
{
    static void Main()
    {
        // Objeto 1 - constructor vacio
        Console.WriteLine("--- Empleado 1 (Vacio) ---");
        Empleado emp1 = new Empleado();
        Console.WriteLine(emp1.ToString());

        // Objeto 2 - constructor con parametros
        Console.WriteLine("--- Empleado 2 (Parametros) ---");
        Empleado emp2 = new Empleado("Donald Trump", 2025, 400000.99);
        Console.WriteLine(emp2.ToString());

        // Parte 2 - Solicitar datos al usuario
        Console.WriteLine("\n--- Empleado 3 (Usuario) ---");
        
        Console.WriteLine("Ingresa el nombre del empleado:");
        string nom = Console.ReadLine();

        Console.WriteLine("Ingrese el año de contratacion:");
        int anio = Convert.ToInt32(Console.ReadLine());

        Console.WriteLine("Ingrese el salario: ");
        double sal = Convert.ToDouble(Console.ReadLine());

        Empleado emp3 = new Empleado(nom, anio, sal);
        Console.WriteLine(emp3.ToString());
    }
}
